server/appConfig/appConfig.py

```diff
@@ -0,0 +1,53 @@
+from ..cxExceptions.cxExceptions import ConfigFileMissingException
+import ConfigParser
+from ..navigation import nav_dict as nav  # Module created by AK to allow dynamic front-end navigation
```

Comment:
I'm exceptionally wary of using modules in other modules (especially our own--security in auth makes sense, though). Can you explain why this is fine or good practice?

Also, if navigation is going to be here, the import in app.py (line 7) doesn't need to be there anymore.

Reply:
The hope is to create small modules that all fit together. These small modules should follow the Single Responsibility Principle (Clean Code p 138 or 169 in the pdf). The big idea behind the SRP is to reduce the complexity of a given change. It also has the nice side effect of reducing the surface area of the code that third party libraries are included in. The hope is that reduced surface area makes updates to third party libraries or swapping those libraries out much less onerous (Clean Code p141 or p144 in the pdf) which in turn helps us avoid shotgun surgery (Refactoring by Robert Martin p 76). 
The hope is that these clean coding practices are language agnostic. And indeed, they helped rapidly speed up debugging at the airport yesterday. That said, neither book uses python as an example, and there may be some implications I've overlooked. 
In order to use our functions in other modules, we'll need to import them somehow. What makes you wary of importing modules?

Reply:
As I understand OOP practices, the idea is modules are--or at least try to be--standalone codebases. From my limited experience, it would seem wise to limit module dependency between them. If the config module was a focal point for initializing most/all modules that need it, though, I could see the practicality of it without it seeming risky.